R/explore_w_histograms.R

```diff
@@ -1,12 +1,97 @@
-#' take a csv file, a string of numerical column names
-#' and returns a png file of histogram(s)
+#' Helper function used to read the file and return a pandas dataframe.
+#' Checks if file type is a .csv or excel. If not, returns a ValueError.
+#'
 #'
-#' @param file csv file path
-#' @param column_names a string of numerical column names, seperate with comma
-#' @return .png file(s) of histogram(s)
+#' @param file : str,the path of the file, including the filetype extension
+#' @param sheet_name: int, default NULL
+#'        if passing an excel file, the name of the sheet to analyze
+#' @return data.frame
 #'
 #' @examples
-#' explore_w_histograms(file = 'toy_dataset.csv', column_names = "volumn,date"])
-explore_w_histograms <- function(file, column_names) {
-  
+#' read_file(file = 'toy_dataset.csv')
+#'
+read_file <- function(file, sheet_name = NULL) {
+  df <- tryCatch({
+    if (tools::file_ext(file) == "csv") {
+      df = readr::read_csv(file)
+    }
+    else {
+      df = readxl::read_xlsx(file, sheet = sheet_name)
+    }
+  },
+  error = function(cond) {
+    message("Please use a valid csv or excel file.")
+    message(cond)
+    return(NA)
+  })
+  return(df)
+}
+
+
+#'Helper function used to take a dataframe, a column name
+#'Returns True if the column is numerical, False otherwise
+#'
+#' @param df: data.frame
+#' @param column: str, the name of the column
+#'
+#' @return logical: TRUE or FALSE
+#'
+#' @examples is_numeric(df,'Age')
+is_numeric <- function(df, column){
+  c_class <- class({{df}} %>% pull({{column}}))
+
+  if (c_class == 'character') {
+    return(FALSE)
+  } else if (c_class == 'complex'){
+    return(FALSE)
+  } else if (c_class == 'logical'){
+    return(FALSE)
+  } else {
+    return(TRUE)
+  }
+}
+
+#' Helper function used to take a dataframe, a numerical column name,
+# and returns a png file of a histogram
+#'
+#' @param df: data.frame
+#' @param column: str, the name of the column
+#'
+#' @return a histogram of the column
+#' @export a png file of the histogram
+#'
+#' @examples make_save_histogram(df, 'Age')
+make_save_histogram <- function(df, column){
+  plot <- ggplot({{df}},
+                 aes(get({{column}})))+geom_bar(stat = 'count')
+  ggsave(paste({{column}}, '_chart.png'))
```

Comment:
I'm not sure if you need the `{{}}` for the df, but I could be wrong. I guess we can see if it passes your tests?

Reply:
The` {{ }}` is to pass the input of the function into it (like the `!!column` bang bang Tiffany mentioned)! For example - if I don't use `{{column}}` here it would tell me `column not found`. I'll add test soon to make sure it works!